solc/jsonCompiler.cpp

```diff
@@ -218,12 +218,13 @@ string compile(StringMap const& _sources, bool _optimize, CStyleReadFileCallback
 		{
 			// Do not taint the internal error list
 			ErrorList formalErrors;
-			if (compiler.prepareFormalAnalysis(&formalErrors))
+			ErrorReporter errorReporter(formalErrors);
+			if (compiler.prepareFormalAnalysis(&errorReporter))
 				output["formal"]["why3"] = compiler.formalTranslation();
-			if (!formalErrors.empty())
+			if (!errorReporter.errors().empty())
```

Comment:
The reason for formal errors is that they are not considered compiler errors which make it "return false". On the other hand: We plan to remove the why3 component anyway, so that would solve it, too :-)